StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -92,38 +89,23 @@ extension EmbeddedPaymentElement {
             savedPaymentMethods: loadResult.savedPaymentMethods,
             customer: configuration.customer,
             incentive: loadResult.elementsSession.incentive,
+            analyticsHelper: analyticsHelper,
             delegate: delegate
         )
     }
-}
-
-// MARK: - EmbeddedPaymentMethodsViewDelegate
-
-extension EmbeddedPaymentElement: EmbeddedPaymentMethodsViewDelegate {
-    func heightDidChange() {
-        delegate?.embeddedPaymentElementDidUpdateHeight(embeddedPaymentElement: self)
-    }
-
-    func updateSelectionState(isNewSelection: Bool) {
-        // Deferring notifying delegate until the exit of this function guarantees the new payment option comes from the new instance of `EmbeddedFormViewController`
-        defer {
-            if isNewSelection {
-                delegate?.embeddedPaymentElementDidUpdatePaymentOption(embeddedPaymentElement: self)
-                if let selection = embeddedPaymentMethodsView.selection {
-                    analyticsHelper.logNewPaymentMethodSelected(paymentMethodTypeIdentifier: selection.analyticsIdentifier)
-                }
-            }
-        }
 
-        guard case let .new(paymentMethodType) = embeddedPaymentMethodsView.selection else {
-            // This can occur when selection is being reset to nothing selected or to a saved payment method, so don't assert.
-            self.formViewController = nil
-            return
+    /// Helper method to inform delegate only if the payment option changed
+    func informDelegateIfPaymentOptionUpdated() {
```

Comment:
There's a footgun here where if you call the delegate method directly instead of calling this helper, the helper will break b/c it won't know the correct last updated payment option. Not sure there's a good way to prevent that in code.